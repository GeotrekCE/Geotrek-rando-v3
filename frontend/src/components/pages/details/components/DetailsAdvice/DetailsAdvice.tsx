import parse from 'html-react-parser';
import { HtmlText } from 'components/pages/details/utils';
import { AlertTriangle } from 'components/Icons/AlertTriangle';

interface DetailsAdviceProps {
  className?: string;
  text: string;
}

export const DetailsAdvice: React.FC<DetailsAdviceProps> = ({ className, text }) => {
  return (
    <div
      id="details_recommandationAdvice"
      className={`py-4 desktop:py-5 px-3 desktop:px-4 flex
      rounded-2xl border-2 border-solid border-greyDarkColored
      ${className ?? ''}`}
    >
      <div className="mr-2 desktop:mr-4 shrink-0 size-6 desktop:size-12 self-start">
        <AlertTriangle />
      </div>
      <div className="text-greyDarkColored desktop:font-bold text-Mobile-C2 desktop:text-P1 my-auto">
        <HtmlText>{parse(text)}</HtmlText>
      </div>
    </div>
  );
};
