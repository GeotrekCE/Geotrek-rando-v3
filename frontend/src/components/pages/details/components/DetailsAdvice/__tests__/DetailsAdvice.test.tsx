import { render } from 'services/testing/reactTestingLibraryWrapper';
import { DetailsAdvice } from '../DetailsAdvice';
describe('DetailsAdvice', () => {
  it('should display a DetailsAdvice', () => {
    const propsAdvice = {
      text: "Pour d'autres informations, se reporter aux recommandations sp&eacute;cifiques de chaque &eacute;tape.",
    };
    const component = render(<DetailsAdvice {...propsAdvice} />);

    expect(component).toMatchSnapshot();
  });
});
