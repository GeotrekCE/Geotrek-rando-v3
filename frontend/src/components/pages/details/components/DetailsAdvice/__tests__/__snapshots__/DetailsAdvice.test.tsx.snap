// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsAdvice should display a DetailsAdvice 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="py-4 desktop:py-5 px-3 desktop:px-4 flex
      rounded-2xl border-2 border-solid border-greyDarkColored
      "
        id="details_recommandationAdvice"
      >
        <div
          class="mr-2 desktop:mr-4 flex-shrink-0 w-6 h-6 desktop:h-12 desktop:w-12 self-start"
        >
          <svg
            fill="none"
            viewBox="0 0 49 49"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M4.247 36.178L21.542 7.85a4.1 4.1 0 013.491-1.928 4.1 4.1 0 013.492 1.928L45.82 36.178c.725 1.233.73 2.751.011 3.988a4.097 4.097 0 01-3.503 2.022H7.739a4.097 4.097 0 01-3.503-2.022 3.941 3.941 0 01.011-3.988zm20.787-23.856c1.767 0 3.2 1.591 3.2 3.555v8.889c0 1.964-1.433 3.556-3.2 3.556-1.768 0-3.2-1.592-3.2-3.556v-8.889c0-1.964 1.432-3.555 3.2-3.555zm3.2 23.466a3.2 3.2 0 11-6.4 0 3.2 3.2 0 016.4 0z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="text-greyDarkColored desktop:font-bold text-Mobile-C2 desktop:text-P1 my-auto"
        >
          <div
            class="sc-bdfBQB hRTanl"
          >
            Pour d'autres informations, se reporter aux recommandations spécifiques de chaque étape.
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="py-4 desktop:py-5 px-3 desktop:px-4 flex
      rounded-2xl border-2 border-solid border-greyDarkColored
      "
      id="details_recommandationAdvice"
    >
      <div
        class="mr-2 desktop:mr-4 flex-shrink-0 w-6 h-6 desktop:h-12 desktop:w-12 self-start"
      >
        <svg
          fill="none"
          viewBox="0 0 49 49"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M4.247 36.178L21.542 7.85a4.1 4.1 0 013.491-1.928 4.1 4.1 0 013.492 1.928L45.82 36.178c.725 1.233.73 2.751.011 3.988a4.097 4.097 0 01-3.503 2.022H7.739a4.097 4.097 0 01-3.503-2.022 3.941 3.941 0 01.011-3.988zm20.787-23.856c1.767 0 3.2 1.591 3.2 3.555v8.889c0 1.964-1.433 3.556-3.2 3.556-1.768 0-3.2-1.592-3.2-3.556v-8.889c0-1.964 1.432-3.555 3.2-3.555zm3.2 23.466a3.2 3.2 0 11-6.4 0 3.2 3.2 0 016.4 0z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="text-greyDarkColored desktop:font-bold text-Mobile-C2 desktop:text-P1 my-auto"
      >
        <div
          class="sc-bdfBQB hRTanl"
        >
          Pour d'autres informations, se reporter aux recommandations spécifiques de chaque étape.
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
